### ISSUE TITLE
Missing Match Arm when using conditional enum variants and `thiserror`

### ISSUE THREAD

**Issue by insertdead** [rust-lang/rust-analyzer#17020]
**rust-analyzer version**: latest `nightly` version as of 2024-04-05 ([source](https://github.com/rust-lang/rust-analyzer/tree/8e581ac348e223488622f4d3003cb2bd412bf27e))

**rustc version**: rustc 1.76.0 (07dca489a 2024-02-04) (built from a source tarball)

**editor or extension**: VSCode with Rust-Analyzer v0.3.1850

**relevant settings**: `rust-analyzer.cargo.features` set to `all`

**repository**: https://github.com/insertdead/ra-missing-match-arm

An enum such as the one below deriving the `Error` proc macro with some feature-dependent variants creates a false positive about missing match arms, despite compiling with no issues and `cargo check` not finding anything.
`thiserror` is the only dependency.

**code snippet to reproduce**: 
```rust
use thiserror::Error;

#[derive(Debug, Error)]
pub enum ShenanigansAfoot {
    #[error("{0}")]
    NormalVariant(String),
    #[error("testing")]
    OtherVariant,
    #[cfg(feature = "feature1")]
    #[error("{0}")]
    FeatureVariant(i32),
    #[cfg(feature = "feature2")]
    #[error("{0}")]
    OtherFeature(u64)
}
```

P.S. I tried to recreate this with serde by deriving `Deserialize` and the error did not reappear. Is it maybe something to do with the `Display` implementation `thiserror` creates?

**Comment by insertdead** [comment]
Hmm, not sure what changed, but it's no longer showing any errors. Going to close for now.
